post_info.php

```diff
@@ -29,6 +29,25 @@
         case 'surveys':
             // Get the id of the survey
             $survey_id = $REQUEST_data['survey_id'];
+            // If the user wants to set the attribute "viewable_by_others", call
+            // the function "handle_viewable_by_others" if the variable is valid
+            if (isset($REQUEST_data['viewable_by_others'])) {
+                $viewable_by_others = $REQUEST_data['viewable_by_others'];
+                // Check the validation of $viewable_by_others
+                if (
+                    !in_array($viewable_by_others, array(
+                        "true",
+                        "false",
+                        "0",
+                        "1"
+                    ))
```

Comment:
This is yucky. I encapsulate it in a function. Also add a comment that `true` will match with `"true"`.